File: 0237-delete-node-in-a-linked-list/0237-delete-node-in-a-linked-list.go
func deleteNode(node *ListNode) {
    node.Val = node.Next.Val
    if node.Next.Next == nil{
        node.Next = nil
    } else {
        deleteNode(node.Next)
    }
}

/**
 * Definition for singly-linked list.
 * type ListNode struct {
 *     Val int
 *     Next *ListNode
 * }
 */

File: 0237-delete-node-in-a-linked-list/0237-delete-node-in-a-linked-list.java
class Solution {
    public void deleteNode(ListNode node) {
        node.val = node.next.val;
        if(node.next.next == null) node.next = null;
        else deleteNode(node.next);
    }
}

/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode(int x) { val = x; }
 * }
 */